### ISSUE TITLE
Can't enable gitgutter with GitGutterEnable after disabling it on startup.

### ISSUE THREAD

**Issue by ZoranLoncarevic** [airblade/vim-gitgutter#409]
I installed gitgutter, and it works fine, but after adding

    let g:gitgutter_enabled = 0

to .vimrc in order to turn gitgutter off on startup, I can no longer turn it on with GitGutterEnable. On the other hand, with

    autocmd VimEnter * let g:gitgutter_enabled = 0

in .vimrc, GitGutterEnable and GitGutterToggle (and gitgutter in general) subsequently work fine.

I am using dein plugin manager, but I am not lazy-loading gitgutter. Since I am using vim 7.4, I also have

    let g:gitgutter_async=0

**Comment by airblade** [comment]
That's strange, `let g:gitgutter_enabled=0` followed by `GitGutterEnable` works for me.

What version of vim 7.4 are you using?

**Comment by ZoranLoncarevic** [comment]
vi --version reports:

    VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jul 28 2016 18:21:44)
    Included patches: 1-2068

**Comment by airblade** [comment]
That should be fine.  I can't think what the problem might be and I still can't reproduce it.

I'm not familiar with dein but is it possible to bypass it and just run vim + vim-gitgutter directly?  That would tell us whether or not the problem is with dein.

**Comment by ZoranLoncarevic** [comment]
I can reproduce the problem even whit no other plugin present. With .vim containing only vim-gitgutter plugin files, and NO .vimrc everything works fine. If I then do

    echo let g:gitgutter_enabled = 0 > ~/.vimrc

gitgutter is inactive when I start vim but now GitGutterEnable does nothing (it does not activate gitgutter, nor does it produce any output).

**Comment by airblade** [comment]
With that minimal vimrc, does vim-gitgutter show up in `:scriptnames`?

**Comment by ZoranLoncarevic** [comment]
Yes, the output of scriptnames contain lines

      7: ~/.vim/plugin/gitgutter.vim
      8: ~/.vim/autoload/gitgutter/highlight.vim
    21: ~/.vim/autoload/gitgutter.vim
    22: ~/.vim/autoload/gitgutter/utility.vim
    23: ~/.vim/autoload/gitgutter/hunk.vim

**Comment by airblade** [comment]
I followed your [instructions](https://github.com/airblade/vim-gitgutter/issues/409#issuecomment-292222798) but `:GitGutterEnable` does activate gitgutter for me.

I have no idea why it doesn't work for you, sorry :(

**Comment by airblade** [comment]
Do you still have this problem?

**Comment by mweitzel** [comment]
I have a similar if not the same issue. In my `.vimrc` I have the lines
```
let g:gitgutter_enabled = 0
let updatetime = 25
```
in a new repository I have a file named `file`, with two lines added in a single commit
```
ggutter-repro$ git log -p                                                                                              
commit 54370a9b49a7a84c365db4e1ec142cdf6f95246b            
Author: Matthew Weitzel <redacted@example.com>               
Date:   Mon Oct 30 10:16:48 2017 -0600                     

    file with two lines      

diff --git a/file b/file     
new file mode 100644         
index 0000000..7bba8c8       
--- /dev/null                
+++ b/file                   
@@ -0,0 +1,2 @@              
+line 1                      
+line 2                      
ggutter-repro$ 
```
Here is a screenshot of two separate vim sessions. 

<img width="194" alt="screen shot 2017-10-30 at 10 23 51 am" src="https://user-images.githubusercontent.com/318925/32182545-96b4b9f2-bd5c-11e7-9e9e-d4f02419dc8a.png">

The `1` at the bottom is the output of `:echo g:gitgutter_enabled`

To create the first first (top) vim session (to reproduce the bug) I edited the file before enabled git gutter:
- `vim file` (with `.vimrc` containing `let g:gitgutter_enabled = 0` and `let updatetime = 25`)
- indented line 2
- `:GitGutterEnable`
- `:echo g:gitgutter_enabled`

The bug does not appear when you enable gitgutter _before_ editting, for the second (bottom) vim session, I:
- `vim file` (with `.vimrc` containing `let g:gitgutter_enabled = 0` and `let updatetime = 25`)
- `:GitGutterEnable`
- indented line 2
- `:echo g:gitgutter_enabled`

---

output to `vim --version`
```
$ vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Sep  3 2017 18:35:35)
MacOS X (unix) version
Included patches: 1-1050
Compiled by Homebrew
Huge version without GUI.  Features included (+) or not (-):
+acl             +file_in_path    +mouse_sgr       +tag_old_static
+arabic          +find_in_path    -mouse_sysmouse  -tag_any_white
+autocmd         +float           +mouse_urxvt     -tcl
-balloon_eval    +folding         +mouse_xterm     +termguicolors
-browse          -footer          +multi_byte      +terminal
++builtin_terms  +fork()          +multi_lang      +terminfo
+byte_offset     -gettext         -mzscheme        +termresponse
+channel         -hangul_input    +netbeans_intg   +textobjects
+cindent         +iconv           +num64           +timers
-clientserver    +insert_expand   +packages        +title
+clipboard       +job             +path_extra      -toolbar
+cmdline_compl   +jumplist        +perl            +user_commands
+cmdline_hist    +keymap          +persistent_undo +vertsplit
+cmdline_info    +lambda          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python          +viminfo
+cscope          +lispindent      -python3         +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con      -lua             +rightleft       +windows
+diff            +menu            +ruby            +writebackup
+digraphs        +mksession       +scrollbind      -X11
-dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     -xim
+emacs_tags      -mouseshape      +startuptime     -xpm
+eval            +mouse_dec       +statusline      -xsmp
+ex_extra        -mouse_gpm       -sun_workshop    -xterm_clipboard
+extra_search    -mouse_jsbterm   +syntax          -xterm_save
+farsi           +mouse_netterm   +tag_binary      
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X_UNIX  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: clang   -L. -fstack-protector -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib  -L/usr/local/lib -o vim        -lncurses -liconv -framework Cocoa   -mmacosx-version-min=10.12 -fstack-protector-strong -L/usr/local/lib  -L/usr/local/Cellar/perl/5.26.0/lib/perl5/5.26.0/darwin-thread-multi-2level/CORE -lperl -lm -lutil -lc -F/usr/local/opt/python/Frameworks -framework Python   -lruby.2.4.1 -lobjc    
```
Plugin installed with vim-8 native plugin, at location `~/.vim/pack/all/start/vim-gitgutter/`

**Comment by mweitzel** [comment]
It looks like `gitgutter#diff#run_diff(1, 0)` fails if `gitgutter#diff#run_diff(0, 0)` is not called first.

**Comment by airblade** [comment]
I think this is fixed by 5bfe5b9 – please could you give it a try?

**Comment by wenijinew** [comment]
This is reproduced in the latest commit: 7b0b509
